### PR TITLE
Look for nomad in GOPATH for tests

### PR DESCRIPTION
**Pull request by achanda**

### CODE REVIEW

File: testutil/server.go [hunk 1]
```diff
@@ -101,8 +101,9 @@ type TestServer struct {
 // NewTestServerConfig creates a new TestServer, and makes a call to
 // an optional callback function to modify the configuration.
 func NewTestServer(t *testing.T, cb ServerConfigCallback) *TestServer {
-	if path, err := exec.LookPath("nomad"); err != nil || path == "" {
-		t.Skip("nomad not found on $PATH, skipping")
+        path := fmt.Sprintf("%s/bin/nomad", os.Getenv("GOPATH"))
```
Comment (dadgar) [inline testutil/server.go]: I think it will be better if you use this: https://github.com/hashicorp/nomad/blob/master/helper/discover/discover.go#L17

Comment (achanda) [inline testutil/server.go]: I assumed that for testing we only want to use the binary in `$GOPATH/bin`. Isn't that true? The discover package looks for `nomad` in `$PWD` too. I'm a beginner in golang, so I might be totally off base here.

**Comment by ryanuber** [comment]
I think we actually should not do this. The testutil package uses your configured $PATH, which can certainly include your $GOPATH if you so choose, but the recommended way of running the tests is using the Makefile, which should automatically build a Nomad binary into /tmp, and alter the $PATH for the test run command.

**Comment by achanda** [comment]
I see your point. Let me close this.

**Pull request closed without merging**